src/parse_policies.c

```diff
@@ -18,22 +18,22 @@ static const int lookup_intervals[] = {
    ['d'] = 1000*60*60*24
 };
 
-int parse_string_to_millisecs(const char *timeStr, int *out){
+static int parse_string_to_millisecs(const char *timeStr, uint64_t *out){
     char should_be_empty;
     unsigned char interval_type;
-    int timeSize;
-    if (sscanf(timeStr, "%d%c%c", &timeSize, &interval_type, &should_be_empty) != 2) {
+    uint64_t timeSize;
+    if (sscanf(timeStr, "%ld%c%c", &timeSize, &interval_type, &should_be_empty) != 2) {
```

Comment:
@ashtul that seems like a use case we don't cover in the tests that should be covered

Reply:
Done!